server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -371,9 +364,7 @@ public enum UpdateSynonymsResultStatus {
         UPDATED
     }
 
-    public record UpdateSynonymsResult(UpdateSynonymsResultStatus updateStatus, ReloadAnalyzersResponse reloadAnalyzersResponse) {}
-
-    public record DeleteSynonymsSetResult(AcknowledgedResponse acknowledgedResponse, ReloadAnalyzersResponse reloadAnalyzersResponse) {}
+    public record SynonymsReloadResult<T>(T synonymsOperationResult, ReloadAnalyzersResponse reloadAnalyzersResponse) {}
```

Comment:
I used a generic SynonymsReloadResult record, to avoid having multiple records that just differ on the response type